### DISCUSSION TITLE
Welcome to Corona_Model Discussions!

### DISCUSSION THREAD

**Discussion by joba-1** [joba-1/Corona_Model#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by joba-1** [comment]
As you can see in my repos here, I like to program, test new things and share what I found. 
And like everyone else on this planet I'd like to know how to best get rid of Covid and Sars CoV-2.

The paper https://t.co/p47HG6OEix?amp=1 has cought my interest because the simulations are based on real data and are easily adjustable even for someone with very low prior knowledge about this like me.

Feel free to ask, contribute, whatever...

**Comment by Maik7** [comment]
Hi,
thanks for your work. This model is very interesting and I like to play with it.
I like to add "vaccine hesitancy".
I also added some plots like deaths and infections relative to overall population. There I got aware, that the number of infections can be near 100% even at the beginning of the simulation:

```
_age limit:  55

max vacc:  0.91

 interactions 
     fraction infected icued   dead
0.1      0.1     **9021**   104  109.0
0.3      0.3     6649    59   94.0
0.5      0.5     4100    46   76.0
0.7      0.7       15     0    0.0
0.9      0.9        2     0    0.0 
.
.
._
print(modeledWorld.number_of_people)
_**10398**_
```

I am confused. 
The same is visible directly in the prepint; I interpreted the plots in the paper as "percentage", but it isn't.
Do you understand, what these numbers mean?

Maik

**Reply by joba-1** [reply]
The numbers in the tables you show are absolute numbers - looks like you start with around 90% initially infected and then infections go down obviously because soon noone can be infected anymore

**Reply by Maik7** [reply]
yes. But it is the same in the pre-print. 
Strange. 
If I interpret the code correctly, the simulation is run for 3000 cycles before vaccination starts. I am trying to reduce these cycles right now...

**Reply by joba-1** [reply]
it runs 3000 cycles in a first test run to know the interactive persons for the 'interactive' mode.
The real simulation then runs again 3000 steps, this time with infections and vaccination turned on

**Reply by Maik7** [reply]
yes, first they do 3000 cycles for the interactive model. This shows, that 3000 cycles are enough to run the pandemic through the whole population. Than the simulation with the different vaccination models is started. They take the first fraction of vaccination (far below herd immunity). And then the run 3000 (!!) cycles. At the end of this 3000 cycles the population is 90% infected.
Can you check your results, which percentage of the population is infected at 0.1 vaccination?

**Comment by joba-1** [comment]
Das sind die Zahlen für 0,1 0,2 0,3 usw.

amount of initially recovered agents: 101
amount of initially recovered agents: 203
amount of initially recovered agents: 304
amount of initially recovered agents: 406
amount of initially recovered agents: 507
amount of initially recovered agents: 609
amount of initially recovered agents: 710

**Reply by Maik7** [reply]
![grafik](https://user-images.githubusercontent.com/83724253/117216311-d4991500-adff-11eb-9ce2-8e7eee6b73c7.png)
ich meinte die Zahl der Infizierten am Ende des ersten Simulationsschrittes (0.1), also wo dieser Plot startet. Auch bei dem originalen Model startet er bei 90%. D. h. das Paper startet mit einer Infektion > Herdenimmunität und betrachtet nur noch den dynamischen Überschwinger.

**Reply by Maik7** [reply]
im Plot nahe 100%, weil 10% schon immun sind

**Reply by joba-1** [reply]
Ich hab das so interpretiert, dass das die Infektionen im Verhältnis zum ungeimpften Verlauf sind. 
Sonst macht das für mich tatsächlich wenig Sinn.
Aber die Wahrheit liegt im Code - und da bin ich noch lang nicht durch. 

Allerdings hab ich mittlerweile festgestellt, dass im Paper (und im Script sim_parallel.py ab Zeile 273)  "mein" Szenario auch schon drin ist (die braune Kurve) - noch mit dem Unterschied, dass dort Kinder < 13 Jahre erst ganz am Schluss berücksichtigt werden. Da sehen die Ergebnisse allerdings nicht so doll aus.

**Reply by Maik7** [reply]
jetzt habe ich verstanden, wie vorgegangen wird. Sie nehmen eine komplett nicht-infizierte Gruppe von etwa 10000 Personen, infizieren 4 Personen und lassen 3000 Simulationsschritte durchlaufen und schauen dann, wie viele wurden infiziert, wie viele Intensivbetten wurden benötigt und wie viele sind gestorben. 
Dann alles auf Start und ein Teil der Bevölkerung wird gemäß Szenario geimpft und dann läuft die Welle erneut durch. Es wird also nicht während der Simulation einer Welle die Impfrate geändert, sondern immer unter konstanten Impfbedingungen ein Ausbruch in einer 10000er Gruppe beobachtet. Hätte das Paper aufmerksamer lesen sollen :-) 
Da ich das nun verstanden habe, lasse ich die Simulation nun unter Berücksichtigung von Impfverweigerern durchlaufen...

**Reply by Maik7** [reply]
So, ist nun durchgelaufen. Die gestrichelten Kurven sind mit 20% Impfverweigerern. 16h Rechenzeit. Mal sehen, ob nicht auch zwei Simulationen parallel rechnen können. Könnte bei 32 GB Ram noch so eben funktionieren.
Zusätzlich habe ich eingebaut, dass die initial infizierten Agenten nie zur geimpften Gruppe gehören, damit immer die gleichen Startbedingungen vorliegen. Bei der Originalversion wurden bei Impffortschritt andere Agenten ausgewählt. Mittelt sich bei vielen Läufen weg, aber ich will ja immer nur einen Lauf vergleichen. 
Ich plotte zusätzlich noch die kummulierten Plots. Da müsste man eigentlich die Läufe mit geringer Impfrate höher gewichten, da die Impfungen mit der Zeit an Fahrt gewinnen.
![grafik](https://user-images.githubusercontent.com/83724253/117478168-18eef700-af5f-11eb-9690-57c34f76c022.png)

**Reply by Maik7** [reply]
Jetzt habe ich noch ergänzt, dass die Impfstoffe bisher erst ab 16, demnächst ab 12 und später vielleicht ab 8 Jahren zugelassen sind und lasse für einzelne Szenarien 100%; 90%, 80% und 70%  Impfbereitschaft durchrechnen.
Weiterhin beginne ich jetzt mit 10 initialen Infektionen auf 10K Einwohner. Das entspricht der Positivrate im Tübinger Modell (wobei die Tests ja nur die Hälfte der Infizierenden finden sollen, so dass die aktuellen Infektionszahlen deutlich höher sein sollten. Aber die Zahlen sollen ja sinken, bevor gelockert werden soll...)

**Reply by Maik7** [reply]
Hier ein interessanter Vergleich. Alle Verteilungen berücksichtigen, dass für Kinder die Impfstoffe erst später zugelassen werden.
Initiale Zulassung ab 16 Jahre
Zulassung ab 12 bei 40% Durchimpfung
Zulassung ab 8 bei 80% Durchimpfung 

Weiterhin beginnt der simulierte Ausbruch mit 10 infizierten. Also 1 auf 1000, das entspricht der Rate der beim Tübinger Modell erkannten Hintergrundaktivität bei einem Inzidenzwert um 50. Da die Tests nur 50% der Infizierenden erkennen, könnte das bei einem Inzidenzwert von 25 realistisch sein.

Keine Impfverweigerer bei den durchgezogenen Kurven.
Impfverweigerer in Höhe von 20% bei den gestrichelten Kurven (d.h. die Kurven enden dann bei 70%. 80% und mehr Durchimpfung geht nur durch Überzeugung der Impfverweigerer.

Bezüglich Zahl der Infektionen nähern sich die verschiedenen Strategien deutlich an. Wenn die Simulation halbwegs realistisch ist, ergibt sich, dass wir möglicherweise keine Herdenimmunität durch Impfen erreichen. Ein erheblicher Teil der Impfverweigerer wird sich in einer finalen Welle (nach Lockerung der Maßnahmen, vermutlich ab Herbst/Winter) infizieren.
Ein sehr großer Unterschied ergibt sich aber bei der Zahl der Verstorbenen. 

Wenn die Simulation zutrifft, hat die Regierung genau die richtige Strategie gewählt und sollte zeitnah die Strategie Richtung "jetzt die Aktiven" ändern. Ich gehe davon aus, dass dazu eine Aufhebung der allgemeinen Impfpriorisierung (ggf. noch Schüler /Studenten vorziehen ?) ausreicht.
![grafik](https://user-images.githubusercontent.com/83724253/117536810-90736380-affd-11eb-899c-53765baf5664.png)

Hier beispielhaft der Verlauf eines Ausbruchs bei 80% Durchimpfung + durchmachte Infektionen; könnte die Situation  sein, wenn alle die wollen und dürfen geimpft sind. Etwa 40% der nicht-geimpften erkrankt dann noch in dieser Simulation.

![grafik](https://user-images.githubusercontent.com/83724253/117536613-4b9afd00-affc-11eb-99b9-ac70ad80c2c2.png)
